subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -150,6 +150,11 @@ public static boolean isRetentionEnabled(Map<String, String> systemPropertiesArg
             .collect(Collectors.toList());
     }
 
+    private static boolean isSystemPropertyEnabled(String systemProperty, Map<String, String> systemPropertiesArgs) {
+        String value = getSystemProperty(systemProperty, systemPropertiesArgs);
+        return value != null && !"false".equalsIgnoreCase(value);
```

Comment:
```suggestion
        return value != null && "true".equalsIgnoreCase(value);
```

I would have expected ☝️ . Did you use `false` for backwards compatibility?

Reply:
Yes. This way anything that just says `-Dorg.gradle.unsafe.vfs.retention` will still enable the feature. The main point is that there's a way to disable it.

Again, we should do this via some general feature-enablement feature where we handle everything properly. For an ad-hoc switch like this I think we're fine.